pkg/skaffold/runner/deployer.go

```diff
@@ -92,6 +92,19 @@ func GetDeployer(ctx context.Context, runCtx *runcontext.RunContext, labeller *l
 				}
 			}
 		}
+		// `helm template` doesn't consistently apply `--namespace` values to rendered resources (see https://github.com/helm/helm/issues/3553).
+		// So we need to provide the helm release namespaces to the `apply` command also.
+		for _, r := range pipelines.Renderers() {
```

Comment:
The real question is what is the user expectation about the `manifests.helm.releases.namespace` property- do they expect the rendered manifests to have this populate the namespace fields. Since helm is itself inconsistent about this, we can probably either remove the namespace field from the render stanza, and only have it in deploy; or, fail if the namespaces are different between the two.

Reply:
Did you have an opinion on which makes more sense here re:

> we can probably either remove the namespace field from the render stanza, and only have it in deploy; or, fail if the namespaces are different between the two.

I think `fail if the namespaces are different between the two` makes sense but don't have a strong opinion

Reply:
According to our latest offline discussion, this comment should be solve with the change of approach to solve this issue.